File: app/src/main/java/com/craft404/maximus/model/GroupMember.java
package com.craft404.maximus.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "GROUP_MEMBERS")
//,foreignKeys = {@ForeignKey(entity = Group.class, parentColumns = "id", childColumns = "groupId"),@ForeignKey(entity = People.class, parentColumns = "id", childColumns = "personId")}
public class GroupMember {

    @PrimaryKey(autoGenerate = true)
    public long id;
    private long groupId;
    private long personId;

    public GroupMember(long groupId, long personId) {
        this.groupId = groupId;
        this.personId = personId;
    }

    public long getId() {
        return id;
    }

    public long getGroupId() {
        return groupId;
    }

    public void setGroupId(long groupId) {
        this.groupId = groupId;
    }

    public long getPersonId() {
        return personId;
    }

    public void setPersonId(long personId) {
        this.personId = personId;
    }
}

File: app/src/main/java/com/craft404/maximus/repository/entity/GroupMember.kt
package com.craft404.maximus.repository.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
    tableName = "group_members",
    foreignKeys = [ForeignKey(
        entity = Group::class,
        onDelete = ForeignKey.CASCADE,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("groupId")
    ), ForeignKey(
        entity = Contact::class,
        onDelete = ForeignKey.CASCADE,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("contactId")
    )],
    indices = [
        Index(
            value = ["groupId"],
            unique = false
        ),
        Index(
            value = ["contactId"],
            unique = false
        )
    ]
)
data class GroupMember(
    val groupId: Long,
    val contactId: Long
) : BaseEntity()